testsuite/features/step_definitions/common_steps.rb

```diff
@@ -197,15 +197,21 @@
 Then(/^the salt event log on server should contain no failures$/) do
   # upload salt event parser log
   file = 'salt_event_parser.py'
-  source = File.dirname(__FILE__) + '/../upload_files/' + file
-  dest = "/tmp/" + file
+  source = "#{File.dirname(__FILE__)}/../upload_files/#{file}"
+  dest = "/tmp/#{file}"
   return_code = file_inject($server, source, dest)
   raise 'File injection failed' unless return_code.zero?
   # print failures from salt event log
   output, _code = $server.run("python3 /tmp/#{file}")
   count_failures = output.to_s.scan(/false/).length
   output = output.join.to_s if output.respond_to?(:join)
-  raise "\nFound #{count_failures} failures in salt event log:\n#{output}\n" if count_failures.nonzero?
+  # Ignore the error if there is only the expected failure from min_salt_lock_packages.feature
+  if count_failures == 1 && !$build_validation
+    ignore_error = output.include?('remove lock')
+  end
+  if count_failures.nonzero?
+    raise "\nFound #{count_failures} failures in salt event log:\n#{output}\n" unless ignore_error
+  end
```

Comment:
bug: if for example `count_failures` is equal to 2, `ignore_error` variable will be undefined

also it light be a bit a bit more readable with `if count_failures.nonzero? and not ignore_error`